src/vs/workbench/contrib/chat/browser/chatSessions.contribution.ts

```diff
@@ -944,36 +946,35 @@ export class ChatSessionsService extends Disposable implements IChatSessionsServ
 				if (state.type !== IChatToolInvocation.StateKind.Completed) {
 					const pastTenseMessage = toolInvocation.pastTenseMessage;
 					const invocationMessage = toolInvocation.invocationMessage;
-					const message = pastTenseMessage || invocationMessage;
-					description = typeof message === 'string' ? message : message?.value ?? '';
+					description = pastTenseMessage || invocationMessage;
 
-					if (description) {
-						description = this.extractFileNameFromLink(description);
-					}
 					if (state.type === IChatToolInvocation.StateKind.WaitingForConfirmation) {
 						const message = toolInvocation.confirmationMessages?.title && (typeof toolInvocation.confirmationMessages.title === 'string'
 							? toolInvocation.confirmationMessages.title
 							: toolInvocation.confirmationMessages.title.value);
-						description = message ?? localize('chat.sessions.description.waitingForConfirmation', "Waiting for confirmation: {0}", description);
+						description = message ?? localize('chat.sessions.description.waitingForConfirmation', "Waiting for confirmation: {0}", typeof description === 'string' ? description : description.value);
 					}
 				}
 			}
 			if (!description && part.kind === 'toolInvocationSerialized') {
-				description = typeof part.invocationMessage === 'string' ? part.invocationMessage : part.invocationMessage?.value || '';
+				description = part.invocationMessage;
 			}
 			if (!description && part.kind === 'progressMessage') {
-				description = part.content.value || '';
+				description = part.content;
 			}
 		}
-
-		return description;
+		return this.cleanUpProgress(description);
 	}
 
-	private extractFileNameFromLink(filePath: string): string {
-		return filePath.replace(/\[(?<linkText>[^\]]*)\]\(file:\/\/\/(?<path>[^)]+)\)/g, (match: string, _p1: string, _p2: string, _offset: number, _string: string, groups?: { linkText?: string; path?: string }) => {
-			const fileName = groups?.path?.split('/').pop() || groups?.path || '';
-			return (groups?.linkText?.trim() || fileName);
-		});
+	private cleanUpProgress(progress: IMarkdownString | string | undefined): string {
+		let cleaned = '';
+		if (typeof progress === 'string') {
```

Comment:
You should be able to pass strings to `renderAsPlaintext` which will return them directly so you don't need this conditional